lib/miq_automation_engine/engine/miq_ae_engine/drb_remote_invoker.rb

```diff
@@ -120,7 +120,16 @@ class AutomateMethodException < StandardError
   # undefine Object#display which would be called over service#display
   DRbObject.send(:undef_method, :display)
 
-  DRb.start_service
+  # DRb.start_service with no URI can attempt to resolve your local hostname[1], which:
+  #   * is slower than just telling it to use a local address/socket
+  #   * could be wrong and in some cases, it can be a remote IP to a DNS assistance program
+  # [1] https://github.com/ruby/ruby/blob/v2_6_5/lib/drb/drb.rb#L879-L884
+  require 'tmpdir'
+  Dir::Tmpname.create("automation_engine_client", nil) do |path|
```

Comment:
Can multiple clients get created simulatneously?  Does there need to be something to tell them apart?

Reply:
Since the client is initiating a connection to the server I don't think the server needs to find a specific client (it couldn't before this change either).  We just need to make sure they don't collide which is what Tmpname is doing for us.

Reply:
Good eye.  I can put `"automation_engine_client_#{Thread.current.object_id}"`.  What do you think?

Reply:
```
irb(main):002:0> t = []; 3.times { t << Thread.new { puts Thread.current.object_id} }; t.join
=> "#<Thread:0x00007fc1b8231988@(irb):2 run>#<Thread:0x00007fc1b8231848@(irb):2 run>#<Thread:0x00007fc1b8231730@(irb):2 run>"
70234997361860
70234997361700
70234997361560
```

Reply:
Will this approach work in podified?  /cc @carbonin

Reply:
oh yeah, each call to `Dir::Tmpname.create` will give a different path already, so it should be fine.

```
irb(main):004:0> Dir::Tmpname.create("automation_engine_client", nil) { |path| puts path }
/var/folders/bf/hh6xb_k15wbfzg4tlbkzch300000gn/T/automation_engine_client20200313-70700-3wl6hd
=> "/var/folders/bf/hh6xb_k15wbfzg4tlbkzch300000gn/T/automation_engine_client20200313-70700-3wl6hd"
irb(main):005:0> Dir::Tmpname.create("automation_engine_client", nil) { |path| puts path }
/var/folders/bf/hh6xb_k15wbfzg4tlbkzch300000gn/T/automation_engine_client20200313-70700-140g3ya
=> "/var/folders/bf/hh6xb_k15wbfzg4tlbkzch300000gn/T/automation_engine_client20200313-70700-140g3ya"
irb(main):006:0> Dir::Tmpname.create("automation_engine_client", nil) { |path| puts path }
/var/folders/bf/hh6xb_k15wbfzg4tlbkzch300000gn/T/automation_engine_client20200313-70700-1b1gyci
=> "/var/folders/bf/hh6xb_k15wbfzg4tlbkzch300000gn/T/automation_engine_client20200313-70700-1b1gyci"
```

Reply:
@jrafanie Turns out Tmpname takes care of it by putting random junk on the end...I misinterpreted the second `nil` (which I think is not needed based on the doc?)

Reply:
> Will this approach work in podified?

Yeah, I believe all the automate Drb is local to the worker. As long as multiple workers are not accessing the same Drb server it will be fine.

Reply:
Yeah, it defaults to `Dir.tmpdir` if `nil`.

That said, if you wanted put this name in `/var/www/miq/vmdb/tmp`, you could then do:

```ruby
Dir::Tmpname.create("automation_engine_client", Rails.root.join("tmp").to_s) { |dir| ... }
```

Reply:
So, are we all good with the current code?   Via, https://github.com/ManageIQ/manageiq-automation_engine/pull/431#discussion_r392345943 it creates random paths each time it's called so each automate "client" that starts a DRb server will get a different path.

Reply:
:+1: